### PR TITLE
add geospatial type support

### LINKED ISSUE #341: Unable to use GeoJSON point as the value of `$nearSphere` operator
It seems the type of `$nearSphere` operator defined in `FilterOperators` only supports legacy coordinate point.

```typescript
interface FilterOperators<TValue> extends Document {
  // [...]
  $nearSphere?: TValue;
  // [...]
}
```

given the type `Place` below

```typescript
interface Place {
  name: string;
  location: [number, number];
}

let placeCollection: Collection<Place> = await getPlaceCollection();
```

#### case 1. GeoJSON point (which is not supported)
```typescript
placeCollection.find(
  {
    location: {
       $nearSphere: { // <-- (?) doesn't work
          $geometry: { 
             type : "Point",
             coordinates : [ -73.9667, 40.78 ]  
          },
          $minDistance: 1000,
          $maxDistance: 5000
       }
    }
  }
)
```

#### case 2. Legacy coordinate point
```typescript
placeCollection.find(
  {
    location: {
       $nearSphere: [ -73.9667, 40.78 ]   // (!) works
    }
  }
)
```

IMO it should be something like this, so it can support GeoJSON point also
```typescript
interface FilterOperators<TValue> extends Document {
  // [...]
  $nearSphere?: TValue | Document;
  // [...]
}
```

##### Links for your conveinence
- [`FilterOperators<T>`](https://deno.land/x/mongo@v0.29.3/src/types.ts#L629)
- [Official Mongo `$nearSphere`](https://www.mongodb.com/docs/manual/reference/operator/query/nearSphere/#-nearsphere)

### PR DESCRIPTION
**Pull request by Jaysok**
## Description
This will add geospatial type support. Thus resolves #341 

## External Dependency
Added external dependency from [@types/geojson](https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/geojson/index.d.ts) which already defined geojson types pretty good, conforming [rfc7946 - GeoJSON Format](https://datatracker.ietf.org/doc/html/rfc7946). 

## Tests
I focused mainly on making a geospatial query easily and make simple GeoJSON queries work with newly added types.
Test data is mostly from MongoDB official documents and I picked some from there.
GeoJSON query objects in the actual queries are picked from the data or made by myself.

* [searching-for-restaurants](https://www.mongodb.com/docs/manual/tutorial/geospatial-tutorial/#searching-for-restaurants)
* [`Place`](https://raw.githubusercontent.com/mongodb/docs-assets/geospatial/restaurants.json)
* [`Neighborhoods`](https://raw.githubusercontent.com/mongodb/docs-assets/geospatial/neighborhoods.json)
* `IPosition` - made by myself

## Possible issues in the future
- Aggregation operator `$geoNear` is not considered.
- MongoDB versioning is not strictly considered.

### CODE REVIEW

**Comment by lucsoft** [comment]
Could you rebase aka fix the merge conflicts from master

**Comment by Jaysok** [comment]
> Could you rebase aka fix the merge conflicts from master

OK, I've just done it

File: deps.ts [hunk 1]
```diff
@@ -6,3 +6,4 @@ export { deferred } from "https://deno.land/std@0.130.0/async/deferred.ts";
 export type { Deferred } from "https://deno.land/std@0.130.0/async/deferred.ts";
 export * as b64 from "https://deno.land/std@0.130.0/encoding/base64.ts";
 export * as hex from "https://deno.land/std@0.130.0/encoding/hex.ts";
+export * as GeoJSON from "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/geojson/index.d.ts";
```
Comment (erfanium) [inline deps.ts:9]: Can you download and save this type as another file in this repo? It's not good that this module to be depended on any external services, only deno.land is allowed

**Review by erfanium — CHANGES_REQUESTED** [review]

File: deps.ts (hunk 1 above)
Comment (Jaysok) [reply deps.ts]: > Can you download and save this type as an another file in this repo? It's not good that this module to be depended on any external services, only deno.land is allowed

Thanks for the review! I've just fixed it.

**Review by erfanium — APPROVED** [review]

**Comment by erfanium** [comment]
@Jaysok Great, thank you!

**Comment by Jaysok** [comment]
> @Jaysok Great, thank you!

Thank you either! @erfanium

**Pull request merged**